lib/tokenize.cpp

```diff
@@ -8296,12 +8296,18 @@ const Token * Tokenizer::findGarbageCode() const
 
     // case keyword must be inside switch
     for (const Token *tok = tokens(); tok; tok = tok->next()) {
-        if (Token::simpleMatch(tok, "switch (") && Token::simpleMatch(tok->linkAt(1), ") {"))
-            tok = tok->linkAt(1)->linkAt(1);
-        else if (tok->str() == "(")
+        if (Token::simpleMatch(tok, "switch (") && Token::simpleMatch(tok->linkAt(1), ")")) {
+            if (tok->linkAt(1)->linkAt(1) && tok->linkAt(1)->linkAt(1)->str() == "}") {
+                tok = tok->linkAt(1)->linkAt(1);
+            } else {
+                while (tok && tok->str() != ";" && tok->str() != "{")
```

Comment:
you are implying that `tok` might become NULL in this loop. If that happens then `tok = tok->next()` will crash. so please tweak the code. if that is possible, if no ";" is found then a "syntax error" would be nice.

Reply:
I have made the nullptr check explicit and added a unit test case for it. Currently the code just returns the token in case tok->next() == nullptr, is that the right thing to do?